source/dub/dub.d

```diff
@@ -1344,7 +1345,7 @@ private class DependencyVersionResolver : DependencyResolver!(Dependency, Depend
 		}
 		auto basepack = pack.basePackage;
 
-		foreach (d; pack.getAllDependencies()) {
+		pack.getAllDependencies((d) {
```

Comment:
Sounds to me that you could / should return an `InputRange` (or maybe `ForwardRange`) from `getAllDependencies` instead of an array.

Reply:
My main worry is that this would be a breaking API change. A new method with a different name would have to be added for this, or at least some kind of struct with an `alias this`  to an array conversion function. I'll look into the latter option.